services/qs-spark-service/src/main/java/ai/quantumics/api/service/impl/AwsConnectionServiceImpl.java

```diff
@@ -162,8 +188,21 @@ public String getFoldersAndFilePath(String bucketName) throws IOException {
 
     @Override
     public String testConnection(String accessMethod) {
-        amazonS3Client = awsCustomConfiguration.amazonS3Client(accessMethod);
-        amazonS3Client.listBuckets();
+        if(isUseConfigBuckets) {
+            if(StringUtils.isEmpty(configBuckets)) {
+                throw new BadRequestException(EMPTY_BUCKET);
+            }
+            List<String> buckets = Arrays.asList(configBuckets.split(SPLIT_COMMA));
+            for(String bucketName : buckets) {
+                AmazonS3 s3Client = awsAdapter.createS3BucketClient(bucketName);
```

Comment:
check only the first bucket .  AmazonS3 s3Client = awsAdapter.createS3BucketClient(bucketName[0]);